Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/view/MultiImageView.java

```diff
@@ -103,45 +105,66 @@
 
     private Context context;
     private ImageView playView;
-    private GestureDetector exoDoubleTapDetector = new GestureDetector(context, new SimpleOnGestureListener() {
-        @Override
-        public boolean onDoubleTap(MotionEvent e) {
-            callback.onDoubleTap();
-            return true;
-        }
-    });
-    private GestureDetector gifDoubleTapDetector = new GestureDetector(context, new SimpleOnGestureListener() {
-        @Override
-        public boolean onDoubleTap(MotionEvent e) {
-            GifDrawable drawable = (GifDrawable) findGifImageView().getDrawable();
-            if (drawable.isPlaying()) {
-                drawable.pause();
-            } else {
-                drawable.start();
-            }
-            return true;
-        }
-
-        @Override
-        public boolean onSingleTapConfirmed(MotionEvent e) {
-            callback.onTap();
-            return true;
-        }
-    });
-
     private PostImage postImage;
     private Callback callback;
     private Mode mode = Mode.UNLOADED;
-
     private boolean hasContent = false;
     private ImageContainer thumbnailRequest;
     private FileCacheDownloader bigImageRequest;
     private FileCacheDownloader gifRequest;
     private FileCacheDownloader videoRequest;
-
     private SimpleExoPlayer exoPlayer;
-
     private boolean backgroundToggle;
+    private boolean imageAlreadySaved = false;
```

Comment:
If you're going to prevent duplicate saves using this, you should probably also set the toolbar item to be greyed out as well.

Reply:
Then I need FileCacheV2 to get merged to use it's toolbarMenuItem.enable() method. I don't want to copy it here from that PR. So this one will have to wait.

Reply:
Wait a minute. This already works. The only more thing that was need to be implemented is to set this flag to true when we download an image from the ImageViewController to avoid double saves.